pkg/plugin/settings.go

```diff
@@ -51,6 +51,9 @@ func LoadSettings(config backend.DataSourceInstanceSettings) (settings Settings,
 	if strings.TrimSpace(settings.QueryTimeout) == "" {
 		settings.QueryTimeout = "60"
 	}
+	if settings.Protocol == "" {
+		settings.Protocol = "native"
+	}
```

Comment:
@yesoreyeram, @scottlepp, I would appreaciate your advice on this change.
Even though it solves the issue, it might not be the best solution, the value from here will never be saved to the `jsonData`.
From what I understand, the current flow is the following:

```mermaid
  graph TD;
      A[client sends request for creating data source to the server]-->B;
      B[server creates DS and responds with empty jsonData]-->C;
      C[client uses empty jsonData on the config page]-->D;
      D[whenever user interacts with the config page, we update jsonData];
```

But if user never interacts with "protocol" radio buttons, protocol will never be put to the `jsonData`, though everything will still work because of the check on the backend. Maybe it would be better to have some optional method for datasources, that returns default config (`jsonData`). So that we can populate the default config on widget creation. In this case we can set protocol to `native` and it will exist in the `jsonData` even if user doesn't touch this setting. I find it weird that in the UI user sees "Native" protocol being set, but actually it's not set at all.
What do you think? Or maybe there is already a way to set default `jsonData` somehow on datasource creation?

Reply:
hmm looks like it was already defaulting to native:  https://github.com/grafana/clickhouse-datasource/blob/main/pkg/plugin/driver.go#L104

Reply:
But `LoadSettings` was called before that: https://github.com/grafana/clickhouse-datasource/blob/main/pkg/plugin/driver.go#L81 and it was failing.

Reply:
Changed the code according to what we've discussed.